### PR TITLE
Fix Http Error logics 

### PR DESCRIPTION
**Pull request by balasubramanian-s**
### Description

Please describe what this change achieves. Ensure you have read the [Contributing to InSpec GCP](https://github.com/inspec/inspec-gcp/CONTRIBUTING.md) document before 
submitting.

### Issues Resolved

List any existing issues this PR resolves, or any Discourse or StackOverflow discussion that's relevant.

Please ensure commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/gcp_backend.rb [hunk 1]
```diff
@@ -277,8 +281,12 @@ def fetch_id(result)
...
   attr_reader :resource_id
 
-  def raise_if_errors(response, err_path, msg_field)
-    errors = self.class.navigate(response, err_path)
+  def raise_if_errors(response, msg_field)
+    errors = if response['error']['message']&.nil?
+               self.class.navigate(response, %w{error errors})
+             else
+               self.class.navigate(response, %w{error message})
+             end
```
Comment (sathish-progress) [inline libraries/gcp_backend.rb:289]:
```suggestion
    errors =  self.class.navigate(response, %w{error message}) || self.class.navigate(response, %w{error errors})
```

File: libraries/gcp_backend.rb [hunk 2]
```diff
@@ -223,7 +223,11 @@ def fetch(base_url, template, var_data, request_type = 'Get', body = nil)
       request_type,
       body,
     )
-    return_if_object get_request.send
+    validate_response get_request.send
+  end
+
+  def validate_response(response)
+    response.is_a?(Net::HTTPNotFound) || response.is_a?(Net::HTTPNoContent) ? nil : return_if_object(response)
```
Comment (sathish-progress) [inline libraries/gcp_backend.rb:230]: do you still need this?

**Review by sathish-progress — CHANGES_REQUESTED** [review]

File: libraries/gcp_backend.rb (hunk 2 above)
Comment (balasubramanian-s) [reply libraries/gcp_backend.rb:230]: yes, for singular resource non-existence usecase we need to return nil to support
 {should_not }  scenarios. Calling return_if_object directly is raising exceptions

Comment (sathish-progress) [reply libraries/gcp_backend.rb:230]: this is not following proper SRP pattern and we need to handle single & plural connections separately, rather here we are handling them all in one big file which is cumbersome.

**Review by sathish-progress — APPROVED** [review]

File: libraries/gcp_backend.rb (hunk 2 above)
Comment (sa-progress) [reply libraries/gcp_backend.rb:230]: @balasubramanian-s return_if_object fails if response type is Array.

**Review by sa-progress — COMMENTED** [review]
We might need to update this PR.

**Pull request merged**